airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -68,40 +70,69 @@ def get_dag_runs(session, dag_id, start_date_gte=None, start_date_lte=None,
+                                             execution_date_lte, start_date_gte, start_date_lte,
+                                             limit, offset)
+
+    return dagrun_collection_schema.dump(DAGRunCollection(dag_runs=dag_run,
+                                                          total_entries=total_entries))
+
+
+def _fetch_dag_runs(query, session, end_date_gte, end_date_lte,
+                    execution_date_gte, execution_date_lte,
+                    start_date_gte, start_date_lte, limit, offset):
+
+    query = _apply_date_filters_to_query(query, end_date_gte, end_date_lte, execution_date_gte,
+                                         execution_date_lte, start_date_gte, start_date_lte)
+    # apply offset and limit
+    dag_run = query.order_by(DagRun.id).offset(offset).limit(limit).all()
+    total_entries = session.query(func.count(DagRun.id)).scalar()
+    return dag_run, total_entries
+
+
+def _apply_date_filters_to_query(query, end_date_gte, end_date_lte, execution_date_gte,
+                                 execution_date_lte, start_date_gte, start_date_lte):
     # filter start date
     if start_date_gte:
         query = query.filter(DagRun.start_date >= start_date_gte)
-
     if start_date_lte:
         query = query.filter(DagRun.start_date <= start_date_lte)
-
     # filter execution date
     if execution_date_gte:
         query = query.filter(DagRun.execution_date >= execution_date_gte)
-
     if execution_date_lte:
         query = query.filter(DagRun.execution_date <= execution_date_lte)
-
     # filter end date
     if end_date_gte:
         query = query.filter(DagRun.end_date >= end_date_gte)
-
     if end_date_lte:
         query = query.filter(DagRun.end_date <= end_date_lte)
+    return query
 
-    # apply offset and limit
-    dag_run = query.order_by(DagRun.id).offset(offset).limit(limit).all()
-    total_entries = session.query(func.count(DagRun.id)).scalar()
-
-    return dagrun_collection_schema.dump(DAGRunCollection(dag_runs=dag_run,
-                                                          total_entries=total_entries))
 
-
-def get_dag_runs_batch():
+@provide_session
+def get_dag_runs_batch(session):
     """
     Get list of DAG Runs
     """
-    raise NotImplementedError("Not implemented yet.")
+    body = request.get_json()
+    try:
+        data = dagruns_batch_form_schema.load(body).data
+    except ValidationError as err:
+        raise BadRequest(detail=err.messages)
+
+    query = session.query(DagRun)
+
+    if data["dag_ids"]:
+        query = query.filter(DagRun.dag_id.in_(data["dag_ids"]))
+
+    dag_runs, total_entries = _fetch_dag_runs(query, session, data["end_date_gte"], data["end_date_lte"],
+                                              data["execution_date_gte"], data["execution_date_lte"],
+                                              data["start_date_gte"], data["start_date_lte"],
+                                              data["page_limit"], data["page_offset"])
```

Comment:
In `api_connexion/parameters.py` we have some hardcoded params like those. Should we extend the list and use them here instead of using strings?

Reply:
It no longer exists but can be added back.

Reply:
Oh indeed, I'm ok with not adding this back